tests/org.eclipse.swt.tests.win32/ManualTests/org/eclipse/swt/tests/win32/snippets/Bug560546_GC_drawString_vs_GC_drawText.java

```diff
@@ -73,39 +80,37 @@ void function(GC gc, int x, int y, String string) {
 
 	static void testSpeeds(Shell shell) {
 		final int finalIterations = 20 * 1000;
-		String report = "";
-
-		Image image = new Image(shell.getDisplay(), 2000, 100);
-		GC gc = new GC(image);
-		for (int isFinalCalc = 0; isFinalCalc < 2; isFinalCalc++) {
-			for (int iTestString = 0; iTestString < testStrings.length; iTestString++) {
-				for (int iTestFunction = 0; iTestFunction < testFunctions.length; iTestFunction++) {
-					TestString testString = testStrings[iTestString];
-					TestFunction testFunction = testFunctions[iTestFunction];
-
-					// Warm up before measuring
-					final int iterations = (isFinalCalc != 0) ? finalIterations : 10;
-
-					final long time1 = System.nanoTime();
-					for (int iIteration = 0; iIteration < iterations; iIteration++) {
-						testFunction.function(gc, 0, 0, testString.string);
-					}
-					final long time2 = System.nanoTime();
-
-					if (isFinalCalc != 0) {
-						final double elapsed = (time2 - time1) / 1000000000.0;
-						report += String.format("%s, %s - %.3f sec\n", testString.caption, testFunction.caption, elapsed);
+		ImageGcDrawer gcDrawer = (gc, width, height) -> {
+			String report = "";
+			for (int isFinalCalc = 0; isFinalCalc < 2; isFinalCalc++) {
+				for (int iTestString = 0; iTestString < testStrings.length; iTestString++) {
+					for (int iTestFunction = 0; iTestFunction < testFunctions.length; iTestFunction++) {
+						TestString testString = testStrings[iTestString];
+						TestFunction testFunction = testFunctions[iTestFunction];
+
+						// Warm up before measuring
+						final int iterations = (isFinalCalc != 0) ? finalIterations : 10;
+
+						final long time1 = System.nanoTime();
+						for (int iIteration = 0; iIteration < iterations; iIteration++) {
+							testFunction.function(gc, 0, 0, testString.string);
+						}
+						final long time2 = System.nanoTime();
+
+						if (isFinalCalc != 0) {
+							final double elapsed = (time2 - time1) / 1000000000.0;
+							report += String.format("%s, %s - %.3f sec\n", testString.caption, testFunction.caption, elapsed);
+						}
 					}
 				}
 			}
-		}
-
-		gc.dispose();
+			MessageBox messageBox = new MessageBox(shell);
+			messageBox.setMessage(report);
+			messageBox.open();
```

Comment:
I don't think opening a message box should happen inside the drawer (i.e., while instantiating the image data), but it should stay where it was before.